### PR TITLE
Chip

### PR DESCRIPTION
**Pull request by julienCarret**

### CODE REVIEW

File: projects/pastanaga-angular/src/lib/chips/chip/chip.component.spec.ts [hunk 1]
```diff
@@ -0,0 +1,74 @@
+import { ChipComponent } from './chip.component';
+import { Component } from '@angular/core';
+import { SpectatorHost } from '@ngneat/spectator';
+import { createHostFactory } from '@ngneat/spectator/jest';
```
Comment (niglesiaszingoni) [inline projects/pastanaga-angular/src/lib/chips/chip/chip.component.spec.ts:4]: you should import `SpectatorHost` from `@ngneat/spectator/jest`

File: projects/pastanaga-angular/src/lib/chips/chip/chip.component.ts [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+    @Input() autoBackgroundAvatar = false;
+
+    @Input() set noCloseButton(value: boolean) {
+        this.canClose = !coerceBooleanProperty(value);
+    }
+
+    canClose = true;
+
+    @Input() disabled = false;
```
Comment (niglesiaszingoni) [inline projects/pastanaga-angular/src/lib/chips/chip/chip.component.ts:21]: it's better to use getters and setters here so you can use `coerseBooleanProperty`

**Review by niglesiaszingoni — APPROVED** [review]
LGTM, just two small comments.

File: projects/demo/src/app/demo/pages/chip-page/chip-page.component.html [hunk 3]
```diff
@@ -0,0 +1,60 @@
+<pa-demo-page>
+    <pa-demo-title>Chip</pa-demo-title>
+    <pa-demo-description>Small and compact chip used to display information, make selections, filter content or input visual content.
```
Comment (mpellerin42) [inline projects/demo/src/app/demo/pages/chip-page/chip-page.component.html:3]: too much unrelated stuff: it's used to display information like text and avatar, and it contain a close/clear button. 
But by itself, this component doesn't make selections, filter or input ;)

File: projects/demo/src/app/demo/pages/chip-page/chip-page.component.html [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+    </pa-demo-description>
+
+    <pa-demo-examples>
+        <pa-chip noCloseButton>Not closable chip</pa-chip>
+        <pa-chip disabled>Disabled chip</pa-chip>
+        <ng-container *ngFor="let chip of chips">
+            <pa-chip *ngIf="chip.contact" [contact]="chip.contact" (closed)="closeChip(chip.name)">{{chip.name}}</pa-chip>
```
Comment (mpellerin42) [inline projects/demo/src/app/demo/pages/chip-page/chip-page.component.html:10]: this component should be agnostic, so the input should not be `contact` as you can pass any kind of text and avatar to the chip, doesn't have to be a user at all.

File: projects/pastanaga-angular/src/lib/chips/chip/chip.component.spec.ts [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+import { MockModule, ngMocks } from 'ng-mocks';
+import { AvatarComponent, ButtonComponent, PaAvatarModule, PaButtonModule } from '../../..';
+import { MockedDebugElement } from 'ng-mocks/dist/lib/mock-render';
```
Comment (ebrehault) [inline projects/pastanaga-angular/src/lib/chips/chip/chip.component.spec.ts:7]: looks like a weird import path, isn't it exported in a proper index.ts?

**Review by ebrehault — APPROVED** [review]

File: projects/demo/src/app/demo/pages/chip-page/chip-page.component.html [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+    <pa-demo-configuration>
+        <div class="pa-demo-configuration-field">
+            <pa-button (click)="addChip()">Add a chip</pa-button>
+        </div>
+        <div class="pa-demo-configuration-field">
+            <pa-button (click)="addAvatarImageChip()">Add an avatar chip with image</pa-button>
+        </div>
+        <div class="pa-demo-configuration-field">
+            <pa-button (click)="addAvatarChip()">Add an avatar chip without image</pa-button>
+        </div>
```
Comment (mpellerin42) [inline projects/demo/src/app/demo/pages/chip-page/chip-page.component.html:23]: simplify your demo: display directly a chip with text only, one with text + close, one with text+avatar+close. you can make the close button as a configurable checkbox if you want, but keep it simple: it's better to see all examples in one go than to have to click on buttons to display more.

File: projects/pastanaga-angular/src/lib/chips/chip/chip.component.html [hunk 7]
```diff
@@ -0,0 +1,21 @@
...
+    <pa-button *ngIf="canClose"
+               class="pa-close-chip-button"
+               icon="cross-large"
+               size="xsmall"
+               kind="secondary"
+               aspect="basic"
+               [disabled]="disabled"
+               (click)="close($event)"
+               (keydown.enter)="close($event)"
+               (keydown.space)="close($event)"></pa-button>
```
Comment (mpellerin42) [inline projects/pastanaga-angular/src/lib/chips/chip/chip.component.html:20]: I don't think you need those two: it will work automatically on if the button has the focus.

File: projects/pastanaga-angular/src/lib/chips/chip/chip.component.ts [hunk 8]
```diff
@@ -0,0 +1,32 @@
...
+import { Observable } from 'rxjs';
+import { coerceBooleanProperty } from '@angular/cdk/coercion';
+
+@Component({
+    selector: 'pa-chip',
+    templateUrl: './chip.component.html',
+    changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class ChipComponent {
+    @Input() contact?: { name: string; image?: Observable<Blob> };
```
Comment (mpellerin42) [inline projects/pastanaga-angular/src/lib/chips/chip/chip.component.ts:11]: split in two inputs: one for the text, one for the image.
I would replace the `ng-content` by the input text, so you can directly use the text passed as avatar's username.

File: projects/pastanaga-angular/src/lib/styles/_chips.scss [hunk 9]
```diff
@@ -0,0 +1,32 @@
...
+        &:hover {
+            background: $color-neutral-secondary-lighter;
+        }
+        &:active {
+            background: $color-neutral-secondary-light;
+        }
+        &.pa-keyboard-focus {
+            background: $color-neutral-secondary-lighter;
+        }
+    }
```
Comment (mpellerin42) [inline projects/pastanaga-angular/src/lib/styles/_chips.scss:21]: to be checked with francisco: are those overrides really necessary?

**Review by mpellerin42 — CHANGES_REQUESTED** [review]

File: projects/demo/src/app/demo/pages/chip-page/chip-page.component.html (hunk 3 above)
Comment (julienCarret) [reply projects/demo/src/app/demo/pages/chip-page/chip-page.component.html:3]: can be "used" to "make selections"... these are basic usages of chips... I can change it :)

File: projects/demo/src/app/demo/pages/chip-page/chip-page.component.html (hunk 4 above)
Comment (julienCarret) [reply projects/demo/src/app/demo/pages/chip-page/chip-page.component.html:10]: so according to my readings about chips... this looks really standard, that's why I made this call.
would you call it nameAndImage ?

File: projects/pastanaga-angular/src/lib/styles/_chips.scss (hunk 9 above)
Comment (mpellerin42) [reply projects/pastanaga-angular/src/lib/styles/_chips.scss:21]: forgot about this comment: the background of the chip is the same as the background of the button so it make sense to override it. Just frustrating.

File: projects/demo/src/app/demo/pages/chip-page/chip-page.component.html (hunk 6 above)
Comment (julienCarret) [reply projects/demo/src/app/demo/pages/chip-page/chip-page.component.html:23]: I don't see your point... if you can remove chips that's just a good idea to add some and avoids the need of reloading the page, where is the problem ?

File: projects/demo/src/app/demo/pages/chip-page/chip-page.component.html [hunk 10]
```diff
@@ -1,37 +1,27 @@
...
     <pa-demo-examples>
         <pa-chip noCloseButton>Not closable chip</pa-chip>
         <pa-chip disabled>Disabled chip</pa-chip>
-        <ng-container *ngFor="let chip of chips">
-            <pa-chip *ngIf="chip.contact" [contact]="chip.contact" (closed)="closeChip(chip.name)">{{chip.name}}</pa-chip>
-            <pa-chip *ngIf="!chip.contact" (closed)="closeChip(chip.name)">{{chip.name}}</pa-chip>
-        </ng-container>
+        <pa-chip *ngIf="first" (closed)="close('first')">Standard chip</pa-chip>
+        <pa-chip *ngIf="second" (closed)="close('second')" [avatar]="avatarWithImage">Chip with avatar</pa-chip>
+        <pa-chip *ngIf="third" (closed)="close('third')" [avatar]="{userName: 'Amanda Caballerot'}">Chip with avatar base on name</pa-chip>
```
Comment (mpellerin42) [inline projects/demo/src/app/demo/pages/chip-page/chip-page.component.html:11]: typo: `based`

File: projects/demo/src/app/demo/pages/chip-page/chip-page.component.html [hunk 11]
```diff
@@ -1,37 +1,27 @@
...
         <h3>Inputs</h3>
 
         <dl>
-            <dt>contact</dt>
-            <dd>Object providing name and image to avatar <code> name: string, image?: Observable of Blob</code>. <small>(optional)</small></dd>
-
-            <dt>autoBackgroundAvatar</dt>
-            <dd>Assign background color automatically ti the avatar<small>(false by default)</small></dd>
+            <dt>avatar</dt>
+            <dd>Configuration of the avatar to display<code>AvatarModel</code>. <small>(optional)</small></dd>
```
Comment (mpellerin42) [inline projects/demo/src/app/demo/pages/chip-page/chip-page.component.html:24]: you forgot a space in `display <code>AvatarModel</code>`

File: projects/demo/src/app/demo/pages/chip-page/chip-page.component.ts [hunk 12]
```diff
@@ -9,37 +9,38 @@ import { markForCheck } from '../../../../../../pastanaga-angular/src';
...
+    third = true;
+
+    readonly avatarWithImage: AvatarModel = {
+        userName: 'Amanda',
+        image: avatar,
+    };
 
     code = `<pa-chip>A chip</pa-chip>
-        <pa-chip [contact]="myContactWithNameAndImage">Another chip</pa-chip>`;
+<pa-chip [avatar]="avatar">Another chip</pa-chip>`;
```
Comment (mpellerin42) [inline projects/demo/src/app/demo/pages/chip-page/chip-page.component.ts:22]: nit: ~Another chip~ -> `A chip with avatar`

**Review by mpellerin42 — COMMENTED** [review]
few typos

**Review by mpellerin42 — APPROVED** [review]
